core/src/main/java/jenkins/install/InstallState.java

```diff
@@ -25,63 +25,88 @@
 
 import javax.annotation.CheckForNull;
 
-import org.kohsuke.accmod.Restricted;
-import org.kohsuke.accmod.restrictions.NoExternalUse;
+import hudson.Extension;
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
+import hudson.security.HudsonPrivateSecurityRealm;
+import jenkins.model.Jenkins;
 
 /**
  * Jenkins install state.
  *
  * @author <a href="mailto:tom.fennelly@gmail.com">tom.fennelly@gmail.com</a>
  */
-@Restricted(NoExternalUse.class)
-public enum InstallState {
+public class InstallState implements ExtensionPoint {
     /**
      * Need InstallState != NEW for tests by default
      */
-    UNKNOWN(true, null),
+    @Extension
```

Comment:
Yes, exactly. I want the instances. It seems to work (and is documented to work this way) @amuniz